topdown/graphql.go

```diff
@@ -85,6 +85,36 @@ func validateQuery(schema *gqlast.Schema, query *gqlast.QueryDocument) error {
 	return nil
 }
 
+func getBuiltinSchema() (*gqlast.SchemaDocument, error) {
+	schema, err := gqlparser.ParseSchema(gqlvalidator.Prelude)
+	if err != nil {
+		return nil, fmt.Errorf("Error in gqlparser Prelude (should be impossible): %s", err.Message)
+	}
+	return schema, nil
+}
+
+// NOTE(philipc): This function expects *validated* schema documents, and will break
+// if you feed it arbitrary structures.
+func mergeSchemaDocuments(docA *gqlast.SchemaDocument, docB *gqlast.SchemaDocument) *gqlast.SchemaDocument {
+	ast := &gqlast.SchemaDocument{}
+	ast.Merge(docA)
+	ast.Merge(docB)
+	return ast
+}
+
+// Converts a SchemaDocument into a Schema object that can be used for validation.
+// NOTE(philipc): The error type here is a gqlerror.Error struct,
+// which requires us to treat it specially, since the returned value
+// will *always* be non-null.
+// See: https://staticcheck.io/docs/checks#SA4023 for details.
+func convertSchema(ast *gqlast.SchemaDocument) (*gqlast.Schema, error) {
+	schema, err := gqlvalidator.ValidateSchemaDocument(ast)
+	if err != nil {
+		return nil, fmt.Errorf("Error in gqlparser SchemaDocument to Schema conversion: %s", err.Message)
```

Comment:
[nit] It doesn't matter too much, but it's common to wrap errors when constructing an error like this:
```suggestion
		return nil, fmt.Errorf("error in gqlparser SchemaDocument to Schema conversion: %w", err)
```

IIRC `err`s returned by gqlparser aren't returned as _error interface type_, but they should still satisfy the interface -- i.e., this should work 🤞 

The difference is that errors included via %w in fmt.Errorf can be unwrapped; as I said, it doesn't matter much here, but we could also just do it.

https://go.dev/blog/go1.13-errors is the canonical source for when golang started supported error wrapping like this.